packages/blade/src/components/Box/styled-props/getStyledProps.ts

```diff
@@ -1,6 +1,82 @@
 import type { StyledProps } from '../BaseBox/types';
 import type { KeysRequired } from '~src/_helpers/types';
 
+/**
+ * The input type to these styled-props utilities can be anything as we can just pass all the props and then get the filtered styled-props out of it
+ */
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type StyledPropsInputType = Record<string, any>;
+
+/**
+ * Removing undefined styled-props because when used with spread operator, undefined can override the earlier defined values.
+
+  * Example: the following code will print display: undefined since it is given priority in spread.
+  * 
+  * ```ts
+  * const styledProps = {
+  *  display: undefined,
+  * }
+  *
+  * const buttonProps = {
+  *  display: 'block',
+  * }
+  *
+  * const styles = {
+  *  ...buttonProps,
+  *  ...styledProps,
+  * } // -> Prints { display: undefined }
+  * ```
+  */
+const removeUndefinedStyledProps = (obj: StyledPropsInputType): StyledPropsInputType => {
+  const onlyDefinedStyledProps: StyledProps = {};
+  for (const k in obj) {
+    if (obj[k as keyof StyledProps] !== undefined) {
+      // @ts-expect-error: complex type error
+      // eslint-disable-next-line @typescript-eslint/no-dynamic-delete
+      onlyDefinedStyledProps[k as keyof StyledProps] = obj[k as keyof StyledProps];
+    }
+  }
+  return onlyDefinedStyledProps;
+};
+
+/**
+ * - Removes the props that are not styled-props
+ * - Returns all the styled props along with undefined values.
+ *
+ * Can be used in generating storybook docs and testing the possible props.
+ *
+ *
+ * **Use `getStyledProps` instead if you're using this for adding styled-props!**
+ */
+const filterStyledProps = (props: StyledPropsInputType): KeysRequired<StyledProps> => {
```

Comment:
Shouldn't `filterStyledProps` only filter the provided props? Implying if I call this function with empty object it should return and empty object. So you can perhaps filter the returned values directly here and remove `undefined` keys?

Reply:
This function is helpful when you want to know what all props styled-props support.

So I can call this function in tests to know which all props styled-props can return or I can use this function in storybook to generate the args table. I can rename it to something. couldn't think of any other name 🤔

Reply:
how about `makeBladeStyledProps`?

Reply:
renamed it to `makeStyledProps`